internal/generator/vector/output/splunk/splunk_sink.toml

```diff
@@ -10,14 +10,33 @@ if err != null {
 	.@timestamp = ts
 }
 
+'''
+[transforms.splunk_hec_metadata]
+type = "remap"
+inputs = ["splunk_hec_timestamp"]
+source = '''
+# Splunk 'source' field detection
+if ._internal.log_type == "infrastructure" && ._internal.log_source == "node" {
+    ._internal.splunk.source = to_string!(._internal.systemd.u.SYSLOG_IDENTIFIER || "")
+}
+if ._internal.log_source == "container" {
+   	._internal.splunk.source = join!([._internal.kubernetes.namespace_name, ._internal.kubernetes.pod_name, ._internal.kubernetes.container_name], "_")
+}
+if ._internal.log_type == "audit" {
+   ._internal.splunk.source = ._internal.log_source
+}
+._internal.splunk.sourcetype = "_json"
```

Comment:
This should map to the type based upon the payloadkey

Reply:
this used if no `payload settings`